api/context/src/main/java/io/opentelemetry/context/propagation/MultiTextMapPropagator.java

```diff
@@ -17,39 +20,133 @@
 final class MultiTextMapPropagator implements TextMapPropagator {
   private final TextMapPropagator[] textPropagators;
   private final Collection<String> allFields;
+  private final boolean stopExtractAfterFirst;
+  private final boolean iterateBackwards;
 
-  MultiTextMapPropagator(List<TextMapPropagator> textPropagators) {
-    this.textPropagators = new TextMapPropagator[textPropagators.size()];
-    textPropagators.toArray(this.textPropagators);
-    this.allFields = Collections.unmodifiableList(getAllFields(this.textPropagators));
+  /** Returns a {@link MultiTextMapPropagator} for the given {@code propagators}. */
+  public static TextMapPropagator create(TextMapPropagator... propagators) {
```

Comment:
these methods don't need to be public, right, since the class itself is not?

Reply:
Sure.  Is the idea to force users to go through the interface `create()` methods then?  Hide all implementations?  Is that the idea?

Reply:
yessir